### PR TITLE
Added Reverb and Filter support for XACT under OpenAL

### PR DESCRIPTION
**Pull request by dellis1972**

### CODE REVIEW

**Comment by dellis1972** [comment]
I think I need to fix the MacOS Build still.

File: MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs [hunk 1]
```diff
@@ -260,14 +308,44 @@ private void PlatformSetVolume(float value)
...
+                case FilterMode.LowPass:
+                filterType = EfxFilterType.Lowpass;
+                break;
+                case FilterMode.HighPass:
+                filterType = EfxFilterType.Highpass;
+                break;
+            }
+            this.filterQ = filterQ;
+            this.frequency = frequency;
+#endif
```
Comment (tomspilman) [inline MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs]: I think you have a misunderstanding of the API here.  `PlatformSetFilter` expects immediate application of the filter settings if the sound instance is playing.

You should be able to play a long running or looping sound effect and modify both the filter and reverb settings interactively without restarting playback.

File: MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs [hunk 2]
```diff
@@ -260,14 +308,44 @@ private void PlatformSetVolume(float value)
 
         internal void PlatformSetReverbMix(float mix)
         {
+#if !MONOMAC
+            if (!OpenALSoundController.Efx.IsInitialized)
+                return;
+            reverb = mix;
+#endif
```
Comment (tomspilman) [inline MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs]: Same as with `PlatformSetFilter`... this should immediately affect a sound that is currently playing.

File: MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs [hunk 3]
```diff
@@ -260,14 +308,44 @@ private void PlatformSetVolume(float value)
...
         }
 
         internal void PlatformClearFilter()
         {
+#if !MONOMAC
+            if (!OpenALSoundController.Efx.IsInitialized)
+                return;
+
+            applyFilter = false;
+#endif
```
Comment (tomspilman) [inline MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs]: Note that this is ok not being immediate...  just needs to clear the filter state before next playback.

File: MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs (hunk 1 above)
Comment (dellis1972) [inline MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs]: ok so if its NOT playing we need to apply it before we do, but if we are we need to apply it immediately?

Comment (tomspilman) [inline MonoGame.Framework/Audio/SoundEffectInstance.OpenAL.cs]: Yes.

**Comment by dellis1972** [comment]
@tomspilman Updated :) 

If we are playing already we apply the effect, otherwise we store the values for the next time we start playing. Hope thats right.

**Comment by tomspilman** [comment]
That looks like it should work... thanks @dellis1972 !

**Pull request merged**